lib/matplotlib/pyplot.py

```diff
@@ -976,8 +976,8 @@ def subplot(*args, **kwargs):
 
     .. note::
 
-       Creating a new subplot with a position which is entirely inside a
-       pre-existing axes will trigger the larger axes to be deleted::
+       Creating a new subplot with axes that overlap pre-existing axes will
+       trigger the pre-existing axes to be deleted::
```

Comment:
Something like "Creating a subplot will delete any pre-existing subplot that overlaps with it." seems clearer (possibly add "(if any)" at the end).